packages/intrepid2/src/Discretization/Basis/Intrepid2_Basis.hpp

```diff
@@ -898,6 +898,13 @@ using BasisPtr = Teuchos::RCP<Basis<DeviceType,OutputType,PointType> >;
                                     ">>> ERROR (Basis::getSubCellRefBasis): this method is not supported or should be overridden accordingly by derived classes.");
     }
 
+    /// create a basis object living on host
+    virtual BasisPtr<typename Kokkos::HostSpace::device_type, OutputValueType, PointValueType>
+    getHostBasis() const {
```

Comment:
This is something I wanted since some time, but it might be hard to implement  for hierarchical basis.

Reply:
This is necessary in my opinion. We have frequent situation that a basis is templated. For computing reference (master) element operations, we probably want the host version.

Reply:
@CamelliaDPG @rppawlo I need this function in the hierarchical components as well to use it in the orientation tools.

Reply:
@kyungjoo-kim @mperego Noted.  Yes, it will be slightly more complex for the tensor bases, but I think it won't be bad.  We will just need to get host versions of the component bases, and construct the host version of the composite basis using those.